viewer/Viewer.h

```diff
@@ -111,23 +124,58 @@ namespace Enki
 		typedef QMapIterator<const std::type_info*, const std::type_info*> ManagedObjectsAliasesMapIterator;
 		ManagedObjectsAliasesMap managedObjectsAliases;
 		
+		typedef std::pair<QString, unsigned int> viewerMessage;
+		std::list<viewerMessage> messageList;
+		QString controlError1;	// dislpayed if user try to move an object in trackball mode : action forbiden to prevent glitchies
+		QString controlError2;	// dislpayed if user try to move camera in trackball mode
+		QString controlHelp;	// dislpayed if user press F1
+
+		struct ExtendedAttributes
+		{
+			bool movableByPicking;
+
+			ExtendedAttributes():movableByPicking(false){};
+		};
+		std::map<PhysicalObject*, ExtendedAttributes> objectExtendedAttributesList;
+
 		bool mouseGrabbed;
 		QPoint mouseGrabPos;
 		double wallsHeight;
-		
-		bool doDumpFrames;
+		//! to know if camera is in trackball mode
+		bool trackballView;
 		int dumpFramesCounter;
 	
+		PhysicalObject *pointedObject, *selectedObject;
+		QVector3D pointedPoint;
+
 	public:
 		ViewerWidget(World *world, QWidget *parent = 0);
 		~ViewerWidget();
 	
+		void addManagedObjectsAlias(const std::type_info* key, const std::type_info* value);
+
+		World* getWorld();
+		QVector3D getPointedPoint();
+		PhysicalObject* getPointedObject();
+		PhysicalObject* getSelectedObject();
+		bool isTrackballActivated();
+		QString getHelpString();
+		bool isMovableByPicking(PhysicalObject* object);
+
+		void setMovableByPicking(PhysicalObject* object, bool movable);
+
+		void timerEvent(QTimerEvent * event);
```

Comment:
Yes putting the timerEvent as a public slot is misleading.

So if you want to put the logic of the timer out of the viewer, which is a sensible decision, I see different possible approaches:

1. Subclass the viewer, and put no code in the `timerEvent()` function, and call the `world->step()` and `updateGL()` from the application.
2. Change the API and decide that it is the role of the application, and remove the timer from the Viewer. We can easily make a subclass of `ViewerWidget` called `TimerViewerWidget` that provides this features.

Actually, I think that 2. is the cleanest, although it breaks the API more, but Enki is not supposed to be released so it's ok.